src/datasources/balances-api/zerion-balances-api.service.ts

```diff
@@ -0,0 +1,155 @@
+      'balances.providers.zerion.apiKey',
+    );
+    this.baseUri = this.configurationService.getOrThrow<string>(
+      'balances.providers.zerion.baseUri',
+    );
+    this.defaultExpirationTimeInSeconds =
+      this.configurationService.getOrThrow<number>(
+        'expirationTimeInSeconds.default',
+      );
+    this.defaultNotFoundExpirationTimeSeconds =
+      this.configurationService.getOrThrow<number>(
+        'expirationTimeInSeconds.notFound.default',
+      );
+    this.chainsConfiguration = this.configurationService.getOrThrow<
+      Record<number, ChainAttributes>
+    >('balances.providers.zerion.chains');
+  }
+
+  async getBalances(args: {
+    chainId: string;
+    safeAddress: string;
+    fiatCode: string;
+  }): Promise<Balance[]> {
+    try {
+      const cacheDir = CacheRouter.getZerionBalancesCacheDir(args);
+      const chainName = this._getChainName(args.chainId);
+      const currency = args.fiatCode.toLowerCase();
+      const url = `${this.baseUri}/v1/wallets/${args.safeAddress}/positions?filter[chain_ids]=${chainName}&currency=${currency}&sort=value`;
+      const { data } = await this.dataSource.get<ZerionBalances>({
+        cacheDir,
+        url,
+        notFoundExpireTimeSeconds: this.defaultNotFoundExpirationTimeSeconds,
+        networkRequest: { headers: { Authorization: `Basic ${this.apiKey}` } },
+        expireTimeSeconds: this.defaultExpirationTimeInSeconds,
+      });
+      return this._mapBalances(chainName, data);
+    } catch (error) {
+      throw new DataSourceError(
+        `Error getting ${args.safeAddress} balances from provider: ${asError(error).message}}`,
+      );
+    }
+  }
+
+  private _mapBalances(
+    chainName: string,
+    zerionBalances: ZerionBalance[],
+  ): Balance[] {
+    return zerionBalances
+      .filter((zb) => zb.attributes.flags.displayable === true)
+      .map((zb) => {
+        const implementation = zb.attributes.fungible_info.implementations.find(
+          (implementation) => implementation.chain_id === chainName,
+        );
+        if (!implementation)
+          throw Error(
+            `Zerion error: ${chainName} implementation not found for balance ${zb.id}`,
+          );
+        const fiatBalance = getNumberString(zb.attributes.value ?? 0);
+        const fiatConversion = getNumberString(zb.attributes.price);
+
+        return {
+          ...(implementation.address === null
+            ? this._mapNativeBalance(zb.attributes)
+            : this._mapErc20Balance(zb.attributes, implementation.address)),
+          fiatBalance,
+          fiatConversion,
+        };
+      });
+  }
+
+  private _mapErc20Balance(
+    zerionBalanceAttributes: ZerionAttributes,
+    tokenAddress: string,
+  ): Erc20Balance {
+    const { fungible_info, quantity } = zerionBalanceAttributes;
+    return {
+      tokenAddress,
+      token: {
+        name: fungible_info.name!,
+        symbol: fungible_info.symbol!,
```

Comment:
Why do we need to assert these?

Reply:
Good catch. Their values are not required (can be null) according to Zerion docs, b it's probably better to use nullish coalescing as for `logoUri`. Let me know if you prefer an alternative solution.

Reply:
Done in [d37d2b9](https://github.com/safe-global/safe-client-gateway/pull/1096/commits/d37d2b9f58b8ed28950623131da0e8f76ce86470)